libdisplay-info: init at 0.1.0

PedroHLC opened the pull request:
###### Description of changes

The first guy to get out from https://github.com/NixOS/nixpkgs/pull/215343 to the world.

###### Things done

Built and tested with the PR mentioned above.

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

PedroHLC commented:
Result of `nixpkgs-review pr 216315` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>libdisplay-info</li>
  </ul>
</details>

superherointj commented on pkgs/top-level/all-packages.nix:
```diff
@@ -21248,6 +21248,8 @@ with pkgs;
 
   libliftoff = callPackage ../development/libraries/libliftoff { };
 
+  libdisplay-info = callPackage ../development/libraries/libdisplay-info { };
```
Does not look to be in alphabetical order.

PedroHLC committed:
libdisplay-info: init at 0.1.0

Release: https://gitlab.freedesktop.org/emersion/libdisplay-info/-/releases/0.1.0

superherointj commented on pkgs/development/libraries/libdisplay-info/default.nix:
```diff
@@ -0,0 +1,41 @@
+
+  buildInputs = [ hwdata ];
+
+  prePatch = ''
```
Patch phase comes after unpack phase and before configure phase.
In this case, please, move `prePatch` to after `src` before `nativeBuildInputs`.

PedroHLC replied:
If some coding-style rule is not in [the manual](https://nixos.org/manual/nixpkgs/unstable/#id-1.6) nor in the [CONTRIBUTING docs](https://github.com/NixOS/nixpkgs/blob/3d53723112c6045bd6534ebeb536056fec5a16f8/CONTRIBUTING.md#submitting-changes), then I don't see reasons to comply.

The biggest reason for me to leave ArchLinux was that the TUs would come up with rules from their mind and argue from whatever favors their taste.

Considering, we have great tools like `nixpkgs-fmt` (that I used in this file). We should just enforce its usage through CI and put all the tasteful rules in there.

Everything that is not documented, open space for conflicting argues and takes contributors and maintainers time in a boomerang during code-review that should focus on code instead of style.

AndersonTorres replied:
I don't believe this case specifically blocks code readability.

SuperSandro2000 commented on pkgs/development/libraries/libdisplay-info/default.nix:
```diff
@@ -0,0 +1,41 @@
+
+  buildInputs = [ hwdata ];
+
+  prePatch = ''
```
Patching files skills always be done in postPatch otherwise patches could not apply.

PedroHLC replied:
Good catch, I'll make sure to fix it.
